src/Hocon/Impl/HoconObject.cs

```diff
@@ -394,7 +395,7 @@ public IHoconElement Clone(IHoconElement newParent)
             var clone = new HoconObject(newParent);
             foreach (var kvp in this)
             {
-                clone.SetField(kvp.Key, kvp.Value);
+                clone.SetField(kvp.Key, kvp.Value.CloneValue(clone) as HoconField);
```

Comment:
We need to clone all child elements as well, to make deep copy of referenced subtree